resource/modules/resource_match.cpp

```diff
@@ -53,6 +53,13 @@ struct match_perf_t {
     double max;                    /* Max match time */
     double mean;                   /* Mean match time */
     double M2;                     /* Welford's algorithm */
+    // Track failed matches
+    uint64_t njobs_failed;
+    uint64_t njobs_reset_failed;
+    double min_failed;
+    double max_failed;
+    double mean_failed;
+    double M2_failed;
```

Comment:
I like this, but reducing the duplication might be nice, maybe:

```suggestion
struct time_stats {
    void add_time(double t) {
        double delta = 0.0f;
        double delta2 = 0.2f;
        njobs++;
        njobs_reset++;
        min = (min > elapse)? elapse : min;
        max = (max < elapse)? elapse : max;
        // Welford's online algorithm for variance
        delta = elapse - mean;
        mean += delta / (double)njobs;
        delta2 = elapse - mean;
        M2 += delta * delta2;
    }
    uint64_t njobs = 0;                /* Total match count */
    uint64_t njobs_reset = 0;          /* Jobs since reset match count */
    double min = std::numeric_limits<double>::max ();                    /* Min match time */
    double max = 0;                    /* Max match time */
    double mean = 0;                   /* Mean match time */
    double M2 = 0;                     /* Welford's algorithm */
};
time_stats succeeded;
time_stats failed;
```

Reply:
Good suggestion. Implemented.